libraries/AP_HAL_PX4/GPIO.cpp

```diff
@@ -256,7 +266,24 @@ void PX4GPIO::write(uint8_t pin, uint8_t value)
 
 void PX4GPIO::toggle(uint8_t pin)
 {
-    write(pin, !read(pin));
+    switch (pin) {
+
+#ifdef CONFIG_ARCH_BOARD_PX4FMU_V4
+        case HAL_GPIO_A_LED_PIN:    // Arming LED
+                ioctl(_led_fd, LED_TOGGLE, LED_RED);
```

Comment:
I can't see where LED_RED is defined.... but you should define it to HAL_GPIO_A_LED_PIN for pixracer so you don't need the switch/case here.. There are also some minor coding style issues: ioctl() here should be one indentation less.